### DISCUSSION TITLE
onedrive_2.4.10-1_amd64.deb can't install on Deepin Linux

### DISCUSSION THREAD

**Discussion by changfranklin** [abraunegg/onedrive#1407] [Q&A] [answered]
HiI 
cannot install onedriver, he told me that I need libgcc-s1 and so on.
May I ask the correct installation method? 
I am Deepin system!
please help

**Comment by abraunegg** [comment, marked as answer]
@changfranklin 
Installation of the 'onedrive' Debian sid or bullseye package on Deepin is not supported.

The *only* way to install the client on your platform is to build from source following the correct Debian process: 
1. Install OS dependencies as per https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntu-18x-ubuntu-19x-ubuntu-20x--debian-9-debian-10---x86_64 
2. Compile and install as per https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#compilation--installation

If you wish to change this situation, please consider becoming the Deepin package maintainer and contribute back to your community.

**Reply by changfranklin** [reply]
Dear abraunegg
```
Configuring Global Azure AD Endpoints
Authorize this app visiting:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&response_type=code&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

Enter the response uri: 
**Invalid uri
Could not initialize the OneDrive API**
```
please help

**Reply by abraunegg** [reply]
@changfranklin 
Please review https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#authorize-the-application-with-your-onedrive-account

Most likely you are not copying the URI as provided by the login to authorise the application.

**Reply by changfranklin** [reply]
Thank you for your support！
What should I do to automatically perform synchronization when booting up?

**Reply by abraunegg** [reply]
@changfranklin 
Please read review the documentation to answer your question

**Comment by changfranklin** [comment]
Dear abraunegg
I tested ONEDRIVE synchronization
If you create a file in Deepin, it will be instantly synchronized to the cloud!
But if you create a file on ONEDRIVE on another MAC computer, Deepin will not sync instantly. After waiting for a long time, it won’t sync?
You must restart the computer or issue an instruction to re-synchronize before it will be synchronized
How to deal with？
please help

**Reply by abraunegg** [reply]
@changfranklin 
> But if you create a file on ONEDRIVE on another MAC computer, Deepin will not sync instantly. After waiting for a long time, it won’t sync?

It will sync.

The issue is that you are potentially waiting long enough based on your configuration. The default wait for downloading new content when using `--monitor` is 300 seconds as per the documentation - please re-read the documentation and re-try your tests.

**Reply by changfranklin** [reply]
Dear abraunegg
Yes will not sync
If it syncs within 10 seconds, the file will appear!
Do you know where the problem lies?
Why the files created on ONEDRIVE on MAC will not be synchronized to Deepin?
Thank you.

**Reply by abraunegg** [reply]
@changfranklin 
> You must restart the computer or issue an instruction to re-synchronize before it will be synchronized
> Do you know where the problem lies?

When using 2 systems to sync data using 1 account, this *usually* works without issue.

This tells me that however you are starting the client on Linux it is *not* persisting / keeping on running .. so it is running probably one time at reboot and exiting - thus not running, thus the files not syncing.

> Do you know where the problem lies?

How are you running the client on the Linux Deepin system? You have compiled from source, however how are you running it?
* manually using `onedrive --synchronize`
* manually using `onedrive --monitor`
* configured automatic operation via systemd / initd (depending on how / what Deepin Linux uses)

What does the following output when the sync is not happening:
* Is the client actually running `sudo ps aufxw | grep onedrive`
* Start doing some debug logging and analysis

This also *could* be due to you using Deepin Linux .... You might have to change your Linux distribution to something a little more mainstream.

**Reply by changfranklin** [reply]
Dear abraunegg
I will test again
I can ask you again. I am Deepin Linux. I want to modify the order in which the system starts the driver! How can I modify it?
Please Help

**Reply by abraunegg** [reply]
@changfranklin 
Please read the documentation (https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service) and ask your Deepin Linux support folk for assistance.
